javascript/javascript1/week3/CactusIO-interactive.js

```diff
@@ -0,0 +1,41 @@
+// Adding an activity
+let activities = [];
+function addActivity(date, activity, duration) {
+  if (
+    typeof date === "string" &&
+    typeof activity === "string" &&
+    typeof duration === "number"
+  ) {
+    const newActivity = { date: date, activity: activity, duration: duration };
+    activities.push(newActivity);
+    console.log(activities);
+  } else {
+    console.log(`Please enter the detail in proper format`);
+  }
+}
+addActivity("23/7-18", "Youtube", 30);
+addActivity("25/7-18", "Netflix", 75);
+addActivity("24/7-18", "Playing Games", 80);
+console.log(activities);
+
+// Show my status
+const durationLimit = 200;
+function showStatus(activities) {
+  if (activities.length === 0) {
+    console.log(`Please enter some activity.`);
+  }
+  let totalDuration = 0;
+  for (let i = 0; i < activities.length; i++) {
+    totalDuration += activities[i].duration;
+  }
```

Comment:
Here you should escape or return the function if there are 0 activities. Right now if it there are 0 activities then it will log out that you should enter some activity and continue down the function. Loop over an empty array and check the totalDuration. If you `return` the function inside the `activities.length === 0` condition it will then stop after you've logged out the message.

Reply:
Noted!! This is the point where I am making mistakes usually. Corrected.